action.yml

```diff
@@ -117,24 +119,36 @@ runs:
             fi
           done <<< "${{ inputs.restore-keys }}"
 
-          # Get the default branch dynamically using GitHub API
-          DEFAULT_BRANCH=$(curl -s -H "Authorization: token ${{ github.token }}" \
-            "https://api.github.com/repos/${{ github.repository }}" | \
-            jq -r '.default_branch')
-
-          # Only allow fallback to supported default branches
-          case "$DEFAULT_BRANCH" in
-            main|master|branch-*)
-              while IFS= read -r line; do
-                if [ -n "$line" ]; then
-                  RESTORE_KEYS="${RESTORE_KEYS}"$'\n'"refs/heads/${DEFAULT_BRANCH}/${line}"
-                fi
-              done <<< "${{ inputs.restore-keys }}"
-              ;;
-            *)
-              echo "::warning::Default branch '$DEFAULT_BRANCH' is not supported for cache fallback. Supported branches: main, master, branch-*"
-              ;;
-          esac
+          # Determine branch to use for fallback restore keys
+          FALLBACK_BRANCH_INPUT="${{ inputs.fallback-branch }}"
+
+          if [[ -n "$FALLBACK_BRANCH_INPUT" ]]; then
+            if [[ "$FALLBACK_BRANCH_INPUT" == refs/heads/* ]]; then
+              FALLBACK_BRANCH_INPUT="${FALLBACK_BRANCH_INPUT#refs/heads/}"
+            fi
```

Comment:
The transformation can be done without the test. It will work even if there is no refs/heads prefix, no?